File: spec/java/src/io/kaitai/struct/spec/TestDefaultEndianExprIsLe.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.DefaultEndianExprIsLe;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestDefaultEndianExprIsLe extends CommonSpec {
    @Test
    public void testDefaultEndianExprIsLe() throws Exception {
        DefaultEndianExprIsLe r = DefaultEndianExprIsLe.fromFile(SRC_DIR + "endian_expr.bin");

        assertEquals(r.docs().get(0).indicator(), new byte[] { 0x49, 0x49 });
        assertEquals(r.docs().get(0).main().someInt(), 0x42);
        assertEquals(r.docs().get(0).main().someIntBe(), 0x42);
        assertEquals(r.docs().get(0).main().someIntLe(), 0x42);

        assertEquals(r.docs().get(1).indicator(), new byte[] { 0x4d, 0x4d });
        assertEquals(r.docs().get(1).main().someInt(), 0x42);
        assertEquals(r.docs().get(1).main().someIntBe(), 0x42);
        assertEquals(r.docs().get(1).main().someIntLe(), 0x42);

        assertEquals(r.docs().get(2).indicator(), new byte[] { 0x58, 0x58 });
        assertEquals(r.docs().get(2).main().someInt(), 0x42);
        assertEquals(r.docs().get(2).main().someIntBe(), 0x42);
        assertEquals(r.docs().get(2).main().someIntLe(), 0x42);
    }
}

File: spec/cpp_stl/test_default_endian_expr_is_le.cpp
#include <boost/test/unit_test.hpp>

#include <default_endian_expr_is_le.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_default_endian_expr_is_le) {
    std::ifstream ifs("src/endian_expr.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    default_endian_expr_is_le_t* r = new default_endian_expr_is_le_t(&ks);

    BOOST_CHECK_EQUAL(r->docs()->at(0)->indicator(), "II");
    BOOST_CHECK_EQUAL(r->docs()->at(0)->main()->some_int(), 0x42);
    BOOST_CHECK_EQUAL(r->docs()->at(0)->main()->some_int_be(), 0x42);
    BOOST_CHECK_EQUAL(r->docs()->at(0)->main()->some_int_le(), 0x42);

    BOOST_CHECK_EQUAL(r->docs()->at(1)->indicator(), "MM");
    BOOST_CHECK_EQUAL(r->docs()->at(1)->main()->some_int(), 0x42);
    BOOST_CHECK_EQUAL(r->docs()->at(1)->main()->some_int_be(), 0x42);
    BOOST_CHECK_EQUAL(r->docs()->at(1)->main()->some_int_le(), 0x42);

    BOOST_CHECK_EQUAL(r->docs()->at(2)->indicator(), "XX");
    BOOST_CHECK_EQUAL(r->docs()->at(2)->main()->some_int(), 0x42);
    BOOST_CHECK_EQUAL(r->docs()->at(2)->main()->some_int_be(), 0x42);
    BOOST_CHECK_EQUAL(r->docs()->at(2)->main()->some_int_le(), 0x42);

    delete r;
}
